### PR TITLE
Add `RequiredDependencies` component

### LINKED ISSUE #3027: Show the required dependency in the documentation
For example, https://armeria.dev/docs/advanced-logging needs:
```
<dependency>
    <groupId>com.linecorp.armeria</groupId>
    <artifactId>armeria-logback</artifactId>
    <version>x.y.z</version>
</dependency>
```

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
It's cumbersome to add the same dependencies for Gradle and Maven respectively, in the documentation.
If we add a React component which shows the dependencies for Gradle and Maven by just defining the dependencies,
it would be a lot easier.

Modifications:
- Add `RequriedDependencies` React component.
- Add required dependencies in the documentation
  - I didn't add one for gRPC and Thrift because
    - we have a link to the example which shows the build script.
    - there are a lot of more stuffs to add except the dependencies so I want to users rather find the example and use it.

Result:
- Close #3027

### CODE REVIEW

**Comment by minwoox** [comment]
image

**Comment by minwoox** [comment]
image

File: site/src/pages/docs/advanced-dropwizard-integration.mdx [hunk 1]
```diff
@@ -24,47 +24,10 @@ application integrates with Armeria, you can leverage the following:
...
-      </dependencyManagement>\n
-      <dependency>
-          <groupId>com.linecorp.armeria</groupId>
-          <artifactId>armeria-dropwizard2</artifactId>
-      </dependency>
-    `}</CodeBlock>
-  </TabPane>
-</Tabs>
+<RequiredDependencies
+  boms={[{"groupId":"com.linecorp.armeria", "artifactId":"armeria-bom"},
```
Comment (ikhoon) [inline site/src/pages/docs/advanced-dropwizard-integration.mdx:28]: Could use key name without quote`"`?
```suggestion
  boms={[{groupId:"com.linecorp.armeria", artifactId:"armeria-bom"},
```

File: site/src/pages/docs/server-thrift.mdx [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Running a Thrift service
 
+<Tip>
+
+Visit [armeria-examples](https://github.com/line/armeria-examples) to find a fully working example.
```
Comment (ikhoon) [inline site/src/pages/docs/server-thrift.mdx:5]: 👍

File: site/src/pages/docs/server-annotated-service.mdx [hunk 3]
```diff
@@ -1432,14 +1432,13 @@ for more information about RPC and HTTP service names, and how to use it.
...
-dependencies {
-    implementation 'com.linecorp.armeria:armeria'
-    implementation 'com.linecorp.armeria:armeria-kotlin'
-}
-```
+<RequiredDependencies
+   boms={[{"groupId":"com.linecorp.armeria", "artifactId":"armeria-bom"}]}
+   dependencies={[{"groupId":"com.linecorp.armeria", "artifactId":"armeria-kotlin"}]} />
+
+Then, define the suspend function:
```
Comment (ikhoon) [inline site/src/pages/docs/server-annotated-service.mdx:1441]: nit: suspending function?
https://kotlinlang.org/docs/reference/coroutines/composing-suspending-functions.html

**Comment by minwoox** [comment]
@ikhoon All fixed. 😉

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox

File: site/src/components/required-dependencies.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  artifactId: string;
+  version?: string;
+}
+
+interface RequiredDependenciesProps {
+  boms?: Dependency[];
+  dependencies: Dependency[];
+}
+
+function groovyBom(boms: Dependency[]) {
```
Comment (trustin) [inline site/src/components/required-dependencies.tsx:19]: groovy -> gradle

File: site/src/pages/docs/advanced-dropwizard-integration.mdx [hunk 5]
```diff
@@ -24,47 +24,10 @@ application integrates with Armeria, you can leverage the following:
...
-          <groupId>com.linecorp.armeria</groupId>
-          <artifactId>armeria-dropwizard2</artifactId>
-      </dependency>
-    `}</CodeBlock>
-  </TabPane>
-</Tabs>
+<RequiredDependencies
+  boms={[{groupId:"com.linecorp.armeria", artifactId:"armeria-bom"},
+         {groupId:"io.dropwizard", artifactId:"dropwizard-bom", version:`${versions['io.dropwizard:dropwizard-core']}`}]}
+  dependencies={[{groupId:"com.linecorp.armeria", artifactId:"armeria-dropwizard2"}]} />
```
Comment (trustin) [inline site/src/pages/docs/advanced-dropwizard-integration.mdx:30]: Global comment: Perhaps better formatting like eslint would?

File: site/src/pages/docs/advanced-logging.mdx [hunk 6]
```diff
@@ -41,7 +47,7 @@ For example, the following configuration:
 </configuration>
 ```
 
-will define an appender called `RCEA` which exports the following:
+The configuration defines an appender called `RCEA` which exports the following:
```
Comment (trustin) [inline site/src/pages/docs/advanced-logging.mdx:50]: The above configuration defines ?

**Review by trustin — COMMENTED** [review]
Excellent work!

File: site/src/pages/docs/advanced-dropwizard-integration.mdx (hunk 5 above)
Comment (minwoox) [reply site/src/pages/docs/advanced-dropwizard-integration.mdx]: All fixed. PTAL. 😄

**Comment by ikhoon** [comment]
Still LGTM!

**Review by trustin — APPROVED** [review]
:+1:

**Pull request merged**